### PR TITLE
[New Rule] Detecting Unsecure Elasticsearch Nodes

### LINKED ISSUE #106: [New Rule] Detecting Unsecure Elasticsearch Nodes
## Description
Elasticsearch breaches are commonly caused by an unsecured Elasticsearch configuration (not using authentication and/or not encrypting communications).

This can be detected through the use of Packetbeat (with the addition of `send_all_headers` in the `packetbeat.yml` file).

This rule identifies traffic to the default Elasticsearch port that is not using TLS and does not have authentication configured - thus making it unsecure.

## Required Info

- **Eventing Sources:**
Packetbeat

- **Target Operating Systems:**
Windows, Linux, macOS

- **Platforms**
Elasticsearch

- **Target ECS Version:** 1.5.0
- **New fields required in ECS for this?** NA
- **Related issues or PRs** NA

## Optional Info
- **References:**
```
event.category:network_traffic AND event.dataset:http AND event.kind:event AND status:OK AND 
type:http AND destination.port:9200 AND network.direction:inbound AND NOT 
http.response.headers.content-type:"image/x-icon" AND NOT _exists_:http.request.headers.authorization
```
* event.category: network data
* event.dataset: HTTP-only traffic (eliminating secure TLS traffic)
* event.kind: an event was generated
* status: HTTP status of `OK` to avoid "the winds of the Internet/scanning"
* type: HTTP-only traffic
* destination.port: default port of `9200` for Elasticsearch
* network.direction: inbound to the Elasticsearch server
* http.response.headers.content-type: eliminating noise from `favicon.ico` and scanning
* http.request.headers.authorization: if this field doesn't exist, then there is no authentication to the Elasticsearch node 

## Example Data
**Without authentication**
image
```
{
  "_index": "packetbeat-7.8.1-2020.07.29-000001",
  "_type": "_doc",
  "_id": "33eu3nMBrrfU-yfrSlSz",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2020-08-11T18:00:30.443Z",
    "client": {
      "bytes": 523,
      "ip": "[redacted]",
      "port": 63170
    },
    "network": {
      "type": "ipv4",
      "transport": "tcp",
      "protocol": "http",
      "direction": "inbound",
      "community_id": "1:6Yyvi+QAXsYpgG96+EkRpu9GO58=",
      "bytes": 967
    },
    "method": "get",
    "query": "GET /",
    "host": {
      "id": "c0080aa9b80d185086279e4468171a46",
      "containerized": false,
      "ip": [
        "10.128.0.36",
        "fe80::4001:aff:fe80:24"
      ],
      "name": "insecure-elastic-test",
      "mac": [
        "42:01:0a:80:00:24"
      ],
      "hostname": "insecure-elastic-test",
      "architecture": "x86_64",
      "os": {
        "name": "CentOS Linux",
        "kernel": "3.10.0-1127.13.1.el7.x86_64",
        "codename": "Core",
        "platform": "centos",
        "version": "7 (Core)",
        "family": "redhat"
      }
    },
    "status": "OK",
    "type": "http",
    "server": {
      "ip": "10.128.0.36",
      "port": 9200,
      "bytes": 444
    },
    "http": {
      "request": {
        "headers": {
          "content-length": 0
        },
        "method": "get",
        "bytes": 523
      },
      "response": {
        "status_code": 200,
        "bytes": 444,
        "body": {
          "bytes": 333
        },
        "headers": {
          "content-length": 333,
          "content-type": "application/json; charset=UTF-8"
        },
        "status_phrase": "ok"
      },
      "version": "1.1"
    },
    "agent": {
      "id": "11235ab4-60ef-4d4a-849c-f86a8158f6e4",
      "name": "insecure-elastic-test",
      "type": "packetbeat",
      "version": "7.8.1",
      "hostname": "insecure-elastic-test",
      "ephemeral_id": "42887f4b-6e7f-4382-8713-96ceb1d2812a"
    },
    "cloud": {
      "instance": {
        "id": "6808227356287633137",
        "name": "insecure-elastic-test"
      },
      "machine": {
        "type": "n1-standard-4"
      },
      "availability_zone": "us-central1-a",
      "project": {
        "id": "elastic-security-dev"
      },
      "provider": "gcp"
    },
    "url": {
      "port": 9200,
      "path": "/",
      "full": "http://[redacted]:9200/",
      "scheme": "http",
      "domain": "[redacted]"
    },
    "event": {
      "duration": 91971000,
      "start": "2020-08-11T18:00:30.443Z",
      "end": "2020-08-11T18:00:30.535Z",
      "kind": "event",
      "category": "network_traffic",
      "dataset": "http"
    },
    "source": {
      "ip": "[redacted]",
      "port": 63170,
      "bytes": 523
    },
    "ecs": {
      "version": "1.5.0"
    },
    "user_agent": {
      "original": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36"
    },
    "destination": {
      "ip": "10.128.0.36",
      "port": 9200,
      "bytes": 444
    }
  },
  "fields": {
    "tls.server_certificate.not_before": [],
    "tls.detailed.client_certificate.not_after": [],
    "file.created": [],
    "process.parent.start": [],
    "tls.server_certificate.not_after": [],
    "event.end": [
      "2020-08-11T18:00:30.535Z"
    ],
    "file.ctime": [],
    "tls.client_certificate.not_after": [],
    "tls.client.not_after": [],
    "event.ingested": [],
    "@timestamp": [
      "2020-08-11T18:00:30.443Z"
    ],
    "tls.client_certificate.not_before": [],
    "file.mtime": [],
    "event.created": [],
    "file.accessed": [],
    "tls.server.not_before": [],
    "package.installed": [],
    "tls.client.not_before": [],
    "tls.detailed.server_certificate.not_after": [],
    "event.start": [
      "2020-08-11T18:00:30.443Z"
    ],
    "tls.server.not_after": [],
    "tls.detailed.server_certificate.not_before": [],
    "process.start": [],
    "tls.detailed.client_certificate.not_before": []
  },
  "highlight": {
    "event.category": [
      "@kibana-highlighted-field@network_traffic@/kibana-highlighted-field@"
    ],
    "type": [
      "@kibana-highlighted-field@http@/kibana-highlighted-field@"
    ],
    "network.direction": [
      "@kibana-highlighted-field@inbound@/kibana-highlighted-field@"
    ],
    "event.kind": [
      "@kibana-highlighted-field@event@/kibana-highlighted-field@"
    ],
    "event.dataset": [
      "@kibana-highlighted-field@http@/kibana-highlighted-field@"
    ],
    "status": [
      "@kibana-highlighted-field@OK@/kibana-highlighted-field@"
    ]
  },
  "sort": [
    1597168830443
  ]
}
```
**With authentication**
image
```
{
  "_index": "packetbeat-7.8.1-2020.07.29-000001",
  "_type": "_doc",
  "_id": "7kev3nMBpYSg_Jq84BZc",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2020-08-11T18:02:13.512Z",
    "method": "get",
    "http": {
      "request": {
        "method": "get",
        "bytes": 523,
        "headers": {
          "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36",
          "content-length": 0,
          "connection": "keep-alive",
          "authorization": "Basic [reacted]",
          "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9",
          "host": "[reacted]:9200",
          "upgrade-insecure-requests": "1",
          "accept-encoding": "gzip, deflate",
          "accept-language": "en-US,en;q=0.9",
          "cache-control": "max-age=0"
        }
      },
      "response": {
        "bytes": 444,
        "body": {
          "bytes": 333
        },
        "headers": {
          "content-length": 333,
          "content-type": "application/json; charset=UTF-8",
          "content-encoding": "gzip"
        },
        "status_phrase": "ok",
        "status_code": 200
      },
      "version": "1.1"
    },
    "type": "http",
    "status": "OK",
    "event": {
      "duration": 2801000,
      "start": "2020-08-11T18:02:13.512Z",
      "end": "2020-08-11T18:02:13.515Z",
      "kind": "event",
      "category": "network_traffic",
      "dataset": "http"
    },
    "destination": {
      "ip": "10.128.0.36",
      "port": 9200,
      "bytes": 444
    },
    "client": {
      "ip": "[reacted]",
      "port": 63381,
      "bytes": 523
    },
    "query": "GET /",
    "user_agent": {
      "original": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36"
    },
    "source": {
      "ip": "[reacted]",
      "port": 63381,
      "bytes": 523
    },
    "server": {
      "ip": "10.128.0.36",
      "port": 9200,
      "bytes": 444
    },
    "ecs": {
      "version": "1.5.0"
    },
    "network": {
      "type": "ipv4",
      "transport": "tcp",
      "protocol": "http",
      "direction": "inbound",
      "community_id": "1:PlNIMB6A70Od5yZ86J2F1fnVe2E=",
      "bytes": 967
    },
    "url": {
      "domain": "[reacted]",
      "port": 9200,
      "path": "/",
      "full": "http://[reacted]:9200/",
      "scheme": "http"
    },
    "host": {
      "name": "insecure-elastic-test",
      "id": "c0080aa9b80d185086279e4468171a46",
      "containerized": false,
      "ip": [
        "10.128.0.36",
        "fe80::4001:aff:fe80:24"
      ],
      "mac": [
        "42:01:0a:80:00:24"
      ],
      "hostname": "insecure-elastic-test",
      "architecture": "x86_64",
      "os": {
        "platform": "centos",
        "version": "7 (Core)",
        "family": "redhat",
        "name": "CentOS Linux",
        "kernel": "3.10.0-1127.13.1.el7.x86_64",
        "codename": "Core"
      }
    },
    "agent": {
      "type": "packetbeat",
      "version": "7.8.1",
      "hostname": "insecure-elastic-test",
      "ephemeral_id": "aa4626b0-83e4-4b78-a8e7-c7fc54e3ce01",
      "id": "11235ab4-60ef-4d4a-849c-f86a8158f6e4",
      "name": "insecure-elastic-test"
    },
    "cloud": {
      "project": {
        "id": "elastic-security-dev"
      },
      "provider": "gcp",
      "instance": {
        "id": "6808227356287633137",
        "name": "insecure-elastic-test"
      },
      "machine": {
        "type": "n1-standard-4"
      },
      "availability_zone": "us-central1-a"
    }
  },
  "fields": {
    "tls.server_certificate.not_before": [],
    "tls.detailed.client_certificate.not_after": [],
    "file.created": [],
    "process.parent.start": [],
    "tls.server_certificate.not_after": [],
    "event.end": [
      "2020-08-11T18:02:13.515Z"
    ],
    "file.ctime": [],
    "tls.client_certificate.not_after": [],
    "tls.client.not_after": [],
    "event.ingested": [],
    "@timestamp": [
      "2020-08-11T18:02:13.512Z"
    ],
    "tls.client_certificate.not_before": [],
    "file.mtime": [],
    "event.created": [],
    "file.accessed": [],
    "tls.server.not_before": [],
    "package.installed": [],
    "tls.client.not_before": [],
    "tls.detailed.server_certificate.not_after": [],
    "event.start": [
      "2020-08-11T18:02:13.512Z"
    ],
    "tls.server.not_after": [],
    "tls.detailed.server_certificate.not_before": [],
    "process.start": [],
    "tls.detailed.client_certificate.not_before": []
  },
  "highlight": {
    "event.category": [
      "@kibana-highlighted-field@network_traffic@/kibana-highlighted-field@"
    ],
    "type": [
      "@kibana-highlighted-field@http@/kibana-highlighted-field@"
    ],
    "network.direction": [
      "@kibana-highlighted-field@inbound@/kibana-highlighted-field@"
    ],
    "event.kind": [
      "@kibana-highlighted-field@event@/kibana-highlighted-field@"
    ],
    "event.dataset": [
      "@kibana-highlighted-field@http@/kibana-highlighted-field@"
    ],
    "status": [
      "@kibana-highlighted-field@OK@/kibana-highlighted-field@"
    ]
  },
  "sort": [
    1597168933512
  ]
}
```

### PR DESCRIPTION
**Pull request by peasead**
## Issues
Resolves #106 

## Summary
This Detection Rule identifies Elasticsearch nodes that do not have Transport Layer Security (TLS) and/or lack
authentication and are accepting inbound network connections over the default Elasticsearch port.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)? Yes

### CODE REVIEW

File: rules/network/initial_access_unsecure_elasticsearch_node.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    "https://www.elastic.co/guide/en/beats/packetbeat/current/packetbeat-http-options.html#_send_all_headers",
+]
+risk_score = 47
+rule_id = "31295df3-277b-4c56-a1fb-84e31b4222a9"
+severity = "medium"
+tags = ["Elastic", "Network"]
+type = "query"
+
+query = '''
+event.category:network_traffic AND event.dataset:http AND event.kind:event AND status:OK AND type:http AND destination.port:9200 AND network.direction:inbound AND NOT http.response.headers.content-type:"image/x-icon" AND NOT _exists_:http.request.headers.authorization
```
Comment (brokensound77) [inline rules/network/initial_access_unsecure_elasticsearch_node.toml:35]: @peasead what is the significance of `content-type:"image/x-icon"` here?

File: rules/network/initial_access_unsecure_elasticsearch_node.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+[metadata]
+creation_date = "2020/08/11"
+ecs_version = ["1.5.0"]
+maturity = "production"
+updated_date = "2020/08/11"
+
+[rule]
+author = ["Elastic"]
+description = """
+This Detection Rule identifies Elasticsearch nodes that do not have Transport Layer Security (TLS), and/or lack
```
Comment (brokensound77) [inline rules/network/initial_access_unsecure_elasticsearch_node.toml:10]:
```suggestion
Identifies Elasticsearch nodes that do not have Transport Layer Security (TLS), and/or lack
```

File: rules/network/initial_access_unsecure_elasticsearch_node.toml (hunk 1 above)
Comment (peasead) [reply rules/network/initial_access_unsecure_elasticsearch_node.toml:35]: When you browse to an Elasticsearch node with a browser, it loads `Favicon.ico`. It returns a `200`, so causes the rule to fire even though no auth was successful (or even attempted).

**Review by devonakerr — APPROVED** [review]

File: rules/network/initial_access_unsecure_elasticsearch_node.toml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+[metadata]
+creation_date = "2020/08/11"
+ecs_version = ["1.5.0"]
```
Comment (bm11100) [inline rules/network/initial_access_unsecure_elasticsearch_node.toml:3]:
```suggestion
ecs_version = ["1.6.0"]
```

**Review by bm11100 — APPROVED** [review]
LGTM 👏

File: rules/network/initial_access_unsecure_elasticsearch_node.toml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    "https://www.elastic.co/guide/en/beats/packetbeat/current/packetbeat-http-options.html#_send_all_headers",
+]
+risk_score = 47
+rule_id = "31295df3-277b-4c56-a1fb-84e31b4222a9"
+severity = "medium"
+tags = ["Elastic", "Network"]
+type = "query"
+
+query = '''
+event.category:network_traffic AND event.dataset:http AND event.kind:event AND status:OK AND type:http AND destination.port:9200 AND network.direction:inbound AND NOT http.response.headers.content-type:"image/x-icon" AND NOT _exists_:http.request.headers.authorization
```
Comment (brokensound77) [inline rules/network/initial_access_unsecure_elasticsearch_node.toml:35]:
```suggestion
event.category:network_traffic AND network.protocol:http AND status:OK AND destination.port:9200 AND network.direction:inbound AND NOT http.response.headers.content-type:"image/x-icon" AND NOT _exists_:http.request.headers.authorization
```

File: rules/network/initial_access_unsecure_elasticsearch_node.toml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+false_positives = [
+    """
+    If you have front-facing proxies that provide authentication and TLS, this rule would need to be tuned to eliminate
+    the source IP address of your reverse-proxy.
+    """,
+]
+index = ["packetbeat-*"]
+language = "lucene"
+license = "Elastic License"
+name = "Unsecure Elasticsearch Node"
```
[the remark this answers is not in the document]
Comment (peasead) [reply rules/network/initial_access_unsecure_elasticsearch_node.toml:22]: I think "Connection to an..." indicates that an egress connection was made to an unsecured Elasticsearch node.

Let me stick "Inbound" in front.

**Review by brokensound77 — APPROVED** [review]
LGTM 👍

**Pull request merged**